packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -75,22 +135,51 @@ export function useGlobalSetting( path, blockName, source = 'all' ) {
 	};
 
 	// Unlike styles settings get inherited from top level settings.
-	const resultWithFallback =
-		getSettingValueForContext( blockName ) ?? getSettingValueForContext();
+	const resultWithFallback = useMemo( () => {
+		if ( propertyPath ) {
+			return (
+				getSettingValueForContext( blockName, propertyPath ) ??
+				getSettingValueForContext( undefined, propertyPath )
+			);
+		}
+		const result = {};
+		VALID_SETTINGS.forEach( ( setting ) => {
+			set(
+				result,
+				setting,
+				getSettingValueForContext( blockName, setting ) ??
+					getSettingValueForContext( undefined, setting )
+			);
+		} );
+		return result;
+	}, [
+		mergedConfig,
+		userConfig,
+		baseConfig,
+		blockName,
+		propertyPath,
+		source,
+	] );
 
 	return [ resultWithFallback, setSetting ];
 }
 
-export function useGlobalStyle( path, blockName, source = 'all' ) {
+export function useGlobalStyle(
+	path,
+	blockName,
+	source = 'all',
+	shouldDecodeEncode = true
```

Comment:
I've added this flag here but ideally I'd like to remove it later, once all the panels that rely on useGlobalStyle are updated to match the behavior of TypographyPanel in the current PR.